scripts/src/test/org/oppia/android/scripts/RegexPatternValidationCheckTest.kt

```diff
@@ -0,0 +1,133 @@
+    tempFolder.newFile("testfiles/data/src/main/TestActivity.kt")
+
+    val exception = assertThrows(Exception::class) {
+      runScript(
+        testDirectoryPath = tempFolder.getRoot().toString() + "/testfiles",
+        allowedDirectories = arrayOf("data")
+      )
+    }
+
+    assertThat(exception).hasMessageThat().contains(ScriptResultConstants.REGEX_CHECKS_FAILED)
+    assertThat(outContent.toString().trim()).isEqualTo(
+      "Filename pattern violation: Activities cannot be placed in the data module\n" +
+        "Prohibited file: [ROOT]/data/src/main/TestActivity.kt"
+    )
+  }
+
+  @Test
+  fun testFileContent_noSupportLibraryImport_fileContentIsCorrect() {
+    tempFolder.newFile("testfiles/TestFile.kt")
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(
+      ScriptResultConstants.REGEX_CHECKS_PASSED
+    )
+  }
+
+  @Test
+  fun testFileContent_supportLibraryImport_fileContentIsNotCorrect() {
+    val prohibitedContent = "import android.support.v7.app"
+    val fileContainsSuppotLibraryImport = tempFolder.newFile("testfiles/TestFile.kt")
+    fileContainsSuppotLibraryImport.writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(ScriptResultConstants.REGEX_CHECKS_FAILED)
+    assertThat(outContent.toString().trim())
+      .isEqualTo(
+        "Prohibited content usage found on line no. 1\n" +
+          "File: [ROOT]/testfiles/TestFile.kt\n" +
+          "Failure message: AndroidX should be used instead of the support library"
+      )
+  }
+
+  @Test
+  fun testFilenameAndContent_useProhibitedFileName_useProhibitedFileContent_multipleFailures() {
+    tempFolder.newFolder("testfiles", "data", "src", "main")
+    val prohibitedFile = tempFolder.newFile(
+      "testfiles/data/src/main/TestActivity.kt"
+    )
+    val prohibitedContent = "import android.support.v7.app"
+    prohibitedFile.writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript(
+        testDirectoryPath = tempFolder.getRoot().toString() + "/testfiles",
+        allowedDirectories = arrayOf("data")
+      )
+    }
+
+    assertThat(exception).hasMessageThat().contains(ScriptResultConstants.REGEX_CHECKS_FAILED)
+    assertThat(outContent.toString().trim()).isEqualTo(
+      "Filename pattern violation: Activities cannot be placed in the data module\n" +
+        "Prohibited file: [ROOT]/data/src/main/TestActivity.kt\n\n" +
+        "Prohibited content usage found on line no. 1\n" +
+        "File: [ROOT]/data/src/main/TestActivity.kt\n" +
+        "Failure message: AndroidX should be used instead of the support library"
+    )
+  }
+
+  /** Helper function which executes the main method of the script. */
+  private fun runScript(
+    testDirectoryPath: String = tempFolder.getRoot().toString(),
+    allowedDirectories: Array<String> = arrayOf("testfiles")
```

Comment:
Suggest passing in a List instead of Array since that's a bit more fitting the contract of the directories list.

You can convert it using ``toTypedArray()``

Reply:
Updated.